interleave.el

```diff
@@ -118,139 +181,146 @@ to PAGE. It narrows the subtree when found."
-        "Scroll down the PDF. Look up for available notes."
-        (interactive)
-        (setq *interleave--page-marker* (pdf-view-current-page))
-        (pdf-view-scroll-down-or-previous-page)
-        (unless (= *interleave--page-marker* (pdf-view-current-page))
-          (interleave--go-to-page-note (pdf-view-current-page))))
-
-      (defun interleave--add-note ()
-        "Add note for the current page. If there are already notes for this page,
+(defun interleave--add-note ()
+  "Add note for the current page. If there are already notes for this page,
 jump to the notes buffer."
-        (interactive)
-        (let ((page (pdf-view-current-page)))
-          (if (interleave--go-to-page-note page)
-              (interleave--switch-to-other-window)
-            (interleave--create-new-note page)))))
-  (progn ; if `pdf-tools' is NOT installed
-    (defun interleave--go-to-next-page ()
-      "Go to the next page in PDF. Look up for available notes."
-      (interactive)
-      (doc-view-next-page)
-      (interleave--go-to-page-note (doc-view-current-page)))
-
-    (defun interleave--go-to-previous-page ()
-      "Go to the previous page in PDF. Look up for available notes."
-      (interactive)
-      (doc-view-previous-page)
-      (interleave--go-to-page-note (doc-view-current-page)))
-
-    (defun interleave--scroll-up ()
-      "Scroll up the PDF. Look up for available notes."
-      (interactive)
-      (setq *interleave--page-marker* (doc-view-current-page))
-      (doc-view-scroll-up-or-next-page)
-      (unless (= *interleave--page-marker* (doc-view-current-page))
-        (interleave--go-to-page-note (doc-view-current-page))))
-
-    (defun interleave--scroll-down ()
-      "Scroll down the PDF. Look up for available notes."
-      (interactive)
-      (setq *interleave--page-marker* (doc-view-current-page))
-      (doc-view-scroll-down-or-previous-page)
-      (unless (= *interleave--page-marker* (doc-view-current-page))
-        (interleave--go-to-page-note (doc-view-current-page))))
-
-    (defun interleave--add-note ()
-      "Add note for the current page. If there are already notes for this page,
-jump to the notes buffer."
-      (interactive)
-      (let ((page (doc-view-current-page)))
-        (if (interleave--go-to-page-note page)
-            (interleave--switch-to-other-window)
-          (interleave--create-new-note page))))))
+  (interactive)
+  (let ((page (funcall interleave--pdf-current-page-fn)))
+    (if (interleave--go-to-page-note page)
+        (interleave--switch-to-org-buffer t)
+      (interleave--create-new-note page))))
+
+(defun interleave--sync-pdf-page-current ()
+  "Synchronize the page in the pdf buffer to be the same as the page in
+the current narrowed down notes view."
+  (interactive)
+  (interleave--switch-to-org-buffer)
+  (let ((pdf-page))
+    (save-excursion
+      (org-narrow-to-subtree)
+      (goto-char (point-min))
+      (search-forward-regexp "Notes for page \\(.*\\)")
```

Comment:
Can you search for the `interleave_page_note` property instead. The headline should be changeable. Thanks.